src/components/displayTable/DataDisplay.tsx

```diff
@@ -61,7 +57,9 @@ const DataDisplay: React.FC<DataDisplayProps> = ({ getData }) => {
           pageCount={Math.ceil(totalItems / itemsPerPage)}
           marginPagesDisplayed={2}
           pageRangeDisplayed={5}
-          onPageChange={({ selected }) => setCurrentPage(selected)}
+          onPageChange={({ selected }) =>
+            setCurrentPage(selected * itemsPerPage)
```

Comment:
can u check through this logic? 

note that "offset" in the get request command is related to "id" number. 

lets say there are 100 IDs.
they need not always be number 1 to 100

they could look like (in ulta order) 100, 99, 96, 92, 91

so in the next page, you should request for previous five IDs starting from 91. and not (100 - 5 = 95)

Reply:
changed it , used Id as lastId and passed it to getData. resolved